src/action_executor.cpp

```diff
@@ -138,12 +140,17 @@ bool ActionExecutor::IsDone(std::string* error) const {
           *error = errors::kNoLandmarksDetected;
         }
         world_->surface_box_landmarks.clear();
-        for (size_t i=0; i<result->landmarks.size(); ++i) {
+        for (size_t i = 0; i < result->landmarks.size(); ++i) {
           msgs::Landmark landmark;
           ProcessSurfaceBox(result->landmarks[i], &landmark);
           world_->surface_box_landmarks.push_back(landmark);
         }
         runtime_viz_.PublishSurfaceBoxes(world_->surface_box_landmarks);
+
+        for (size_t i = 0; i < result->surfaces.size(); ++i) {
+          moveit_msgs::CollisionObject surface = result->surfaces[i];
```

Comment:
Once you change the API such that the surface is of a different message type, you will need to pack it into a CollisionObject here. In other words, you will move the code you wrote in surface_segmentation_action.cpp to here.